src/cargo/ops/cargo_add/mod.rs

```diff
@@ -529,27 +532,86 @@ fn get_latest_dependency(
                         break res?;
                     }
                     std::task::Poll::Pending => registry.block_until_ready()?,
                 }
             };
-            let latest = possibilities
-                .iter()
-                .max_by_key(|s| {
-                    // Fallback to a pre-release if no official release is available by sorting them as
-                    // less.
-                    let stable = s.version().pre.is_empty();
-                    (stable, s.version())
-                })
-                .ok_or_else(|| {
-                    anyhow::format_err!(
-                        "the crate `{dependency}` could not be found in registry index."
-                    )
-                })?;
+
+            possibilities.sort_by_key(|s| {
+                // Fallback to a pre-release if no official release is available by sorting them as
+                // less.
+                let stable = s.version().pre.is_empty();
+                (stable, s.version().clone())
+            });
+
+            let mut latest = possibilities.last().ok_or_else(|| {
+                anyhow::format_err!(
+                    "the crate `{dependency}` could not be found in registry index."
+                )
+            })?;
+
+            if config.cli_unstable().msrv_policy {
+                if let Some((req_major, req_minor, req_patch)) =
+                    spec.rust_version().map(parse_rust_version)
+                {
+                    let rust_versions = possibilities
+                        .iter()
+                        .map(|s| {
+                            let rust_version = s.rust_version().map(parse_rust_version);
+                            (s, rust_version)
+                        })
+                        .collect::<Vec<_>>();
+
+                    // Find the latest version of the dep which has a compatible rust-version,
+                    // treating candidates without a rust-version as compatible.
+                    let (latest_msrv, _) = rust_versions
+                        .iter()
+                        .filter(|(_, v)| {
+                            v.map(|(major, minor, patch)| {
+                                // Determine whether or not the rust-versions are compatible by
+                                // comparing the lowest possible versions each of them could
+                                // represent.
+                                (req_major, req_minor, req_patch.unwrap_or(0))
+                                    >= (major, minor, patch.unwrap_or(0))
+                            })
+                            .unwrap_or(true)
+                        })
+                        .last()
+                        .ok_or_else(|| {
+                            let mut error_msg = format!(
+                                "could not find version of crate `{dependency}` that satisfies \
+                                 this package's rust-version"
```

Comment:
Could we say what the current package's rust version is?  I know the user can just look it up in their file but its nice when they can see and consider the problem without doing so